app/data_models/questionnaire_store.py

```diff
@@ -60,30 +63,54 @@ def set_metadata(self, to_set: MutableMapping) -> QuestionnaireStore:
 
         return self
 
-    def set_supplementary_data(self, to_set: MutableMapping) -> None:
+    def set_supplementary_data(
+        self,
+        *,
+        to_set: MutableMapping,
+        base_questionnaire_store_updater: BaseQuestionnaireStoreUpdater,
+    ) -> None:
```

Comment:
We should avoid this circular dependency, this will complicate this further in future. It's odd to pass the QSU which in charge of updating the QS to QS itself.

Worth having a discussion as it might be complex to explain here.

Reply:
Great point, this is way better, done 👍 
Moved the tests to the test file for the QSUB as well